lib/influxer/metrics/metrics.rb

```diff
@@ -162,10 +170,50 @@ def tags
       @attributes.select { |k, _| tag_names.include?(k.to_s) }
     end
 
-    attributes :timestamp
+    def time(parse = false)
+      return unless @time
+
+      return parse_timestamp(@time) if parse
+      @time
+    end
+
+    def time=(val)
+      if val.is_a?(Date) || val.is_a?(Time)
+        @time = parse_time(val)
+      else
+        @time
+      end
+    end
 
     private
 
+    def parse_timestamp(val)
+      case client.time_precision
+      when 'ms'
```

Comment:
The first return case is unnecessary, falsey value ignored by _writer_.

The second one would raise error if timestamp is a Date. And there is also problems with strings.

BTW, default precision for Influxer is 'ms'.